com.unity.netcode.gameobjects/Runtime/Spawning/INetworkInstantiationPayloadSynchronizer.cs

```diff
@@ -0,0 +1,22 @@
+namespace Unity.Netcode
+{
+    /// <summary>
+    /// Interface for synchronizing custom instantiation payloads during NetworkObject spawning.
+    /// Used alongside <see cref="INetworkPrefabInstanceHandler"/> to extend instantiation behavior.
+    /// </summary>
+    public interface INetworkInstantiationPayloadSynchronizer
+    {
+        /// <summary>
+        /// Provides a method for synchronizing instantiation payload data during the spawn process.
+        /// Extends <see cref="INetworkPrefabInstanceHandler"/> to allow passing additional data prior to instantiation
+        /// to help identify or configure the local object instance that should be linked to the spawned NetworkObject.
+        ///
+        /// This method is invoked immediately before <see cref="INetworkPrefabInstanceHandler.Instantiate"/> is called,
+        /// allowing you to cache or prepare information needed during instantiation.
+        /// </summary>
+        /// <param name="serializer">The buffer serializer used to read or write custom instantiation data.</param>
+        void OnSynchronize<T>(ref BufferSerializer<T> serializer) where T : IReaderWriter;
```

Comment:
To avoid confusion with NetworkBehaviours, could we rename this to something more descriptive like `OnInstantiation()`?

Reply:
Sounds good to me! As long as the name clearly reflects that this method is solely responsible for synchronizing data prior to instantiation, I’m happy to update it.

OnBeforeInstantiation works well for that, as long as we keep in mind that it is actually serializing and deserializing data

Reply:
We might rename the interface to something like:
`INetworkPrefabPreInstantianceDataSerializer` `INetworkPrefabPreInstantianceSynchronizer` `INetworkPrefabPreInstantiantiateHandler`
And the method could be:
`OnPreInstanceSerialization`, `SynchronizePreInstance`, `PreInstanceSynchronization`, `HandlePreInstantiateData`

This way, the naming makes it clearer that its about the data flow before instantiation, not the instantiation itself, and it also avoids confusion with NetworkBehaviour.OnSynchronize()

Let me know if any of these seem closer to NGO’s naming conventions, or if you would prefer a shorter variation